addons/project/views/project_portal_project_task_templates.xml

```diff
@@ -34,35 +34,34 @@
-                            </th>
-                            <th t-if="groupby == 'stage_id'" t-attf-colspan="{{grouped_tasks_colspan}}">
-                                <!-- This div is necessary for documents_project_sale -->
-                                <div name="stage_name" class="d-flex w-100 align-items-center">
-                                    <span class="text-truncate" t-field="tasks[0].sudo().stage_id.name"/>
-                                </div>
-                            </th>
-                            <th t-if="groupby == 'priority'" t-attf-colspan="{{grouped_tasks_colspan}}">
-                                <span class="text-truncate" t-field="tasks[0].sudo().priority"/></th>
-                            <th t-if="groupby == 'state'" t-attf-colspan="{{grouped_tasks_colspan}}">
-                                <span class="text-truncate" t-field="tasks[0].sudo().state"/></th>
-                            <th t-if="groupby == 'partner_id'" t-attf-colspan="{{grouped_tasks_colspan}}">
-                                <span t-if="tasks[0].sudo().partner_id"
-                                      class="text-truncate"
-                                      t-field="tasks[0].sudo().partner_id.name"/>
-                                <span t-else="">No Customer</span>
+                                </t>
+                                <t t-if="groupby == 'milestone_id'">
+                                    <span t-if="tasks[0].sudo().milestone_id and tasks[0].sudo().allow_milestones"
+                                        t-field="tasks[0].sudo().milestone_id.name"/>
+                                    <span t-else="">No Milestone</span>
+                                </t>
+                                <t t-if="groupby == 'stage_id'">
+                                    <span class="d-flex w-100 align-items-center" t-field="tasks[0].sudo().stage_id.name"/>
+                                </t>
+                                <t t-if="groupby == 'priority'">
+                                    <span t-field="tasks[0].sudo().priority"/>
+                                </t>
+                                <t t-if="groupby == 'state'">
+                                    <span t-field="tasks[0].sudo().state"/>
+                                </t>
+                                <t t-if="groupby == 'partner_id'">
+                                    <span t-if="tasks[0].sudo().partner_id"
+                                        t-field="tasks[0].sudo().partner_id.name"/>
+                                    <span t-else="">No Customer</span>
+                                </t>
                             </th>
+                            <!-- This space is required in documents_project and hr_timesheet to display the exact position of total hours and documents -->
+                            <th name="extra_stage_space" t-attf-colspan="2"></th>
```

Comment:
Why not altering `colspan_project_tasks` instead of adding a new th just for those 2 modules?

If it is needed to add a th for those modules, why not adding it in those modules instead of directly here. 🤷🏼‍♂️